verticapy/performance/vertica/collection/collection_tables.py

```diff
@@ -0,0 +1,584 @@
+        raise NotImplementedError(f"Drop table not implemented yet")
+
+    # Recall: abstract methods won't raise by default
+    @abstractmethod
+    def get_create_table_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_table_sql is not implemented in the base class CollectionTable."
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def get_create_projection_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def has_copy_staging(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def copy_from_local_file(self, filename: str) -> str:
+        # This method will generate and run the copy statement
+        # It will copy to a staging table when necessary
+        raise NotImplementedError(f"")
+
+
+class AllTableNames(Enum):
+    COLLECTION_EVENTS = "collection_events"
+    COLLECTION_INFO = "collection_info"
+    DC_EXPLAIN_PLANS = "dc_explain_plans"
+    DC_QUERY_EXECUTIONS = "dc_query_executions"
+    DC_REQUESTS_ISSUED = "dc_requests_issued"
+    EXECUTION_ENGINE_PROFILES = "execution_engine_profiles"
+    EXPORT_EVENTS = "export_events"
+    HOST_RESOURCES = "host_resources"
+    QUERY_CONSUMPTION = "query_consumption"
+    QUERY_PLAN_PROFILES = "query_plan_profiles"
+    QUERY_PROFILES = "query_profiles"
+    RESOURCE_POOL_STATUS = "resource_pool_status"
+
+
+def getAllCollectionTables(
+    target_schema: str, key: str
+) -> Mapping[str, CollectionTable]:
+    result = {}
+    for name in AllTableNames:
+        name_str = name.value
+        c = collectionTableFactory(name_str, target_schema, key)
+        result[name.name] = c
+
+    return result
+
+
+def collectionTableFactory(
+    table_name: str, target_schema: str, key: str
+) -> CollectionTable:
+    if table_name == AllTableNames.COLLECTION_EVENTS.value:
+        return CollectionEventsTable(target_schema, key)
+    if table_name == AllTableNames.COLLECTION_INFO.value:
+        return CollectionInfoTable(target_schema, key)
+    if table_name == AllTableNames.DC_EXPLAIN_PLANS.value:
+        return DCExplainPlansTable(target_schema, key)
+    if table_name == AllTableNames.DC_QUERY_EXECUTIONS.value:
+        return DCQueryExecutionsTable(target_schema, key)
+    if table_name == AllTableNames.DC_REQUESTS_ISSUED.value:
+        return DCRequestsIssuedTable(target_schema, key)
+    if table_name == AllTableNames.EXECUTION_ENGINE_PROFILES.value:
+        return DCExecutionEngineProfilesTable(target_schema, key)
+
+    # TODO: eventually this will be an error
```

Comment:
Why it is not an error right now?

Reply:
It's not an error because I have the full list of tables, but I don't have subclasses for each table type (yet). I will add them in a subsequent PR.